### ISSUE TITLE
Batch payload sending

### ISSUE THREAD

**Issue by latos** [rollbar/rollbar-java#30]
Payload sending is currently not batched.  Every event is sent as a separate http request, with a fresh connection.  This seems quite gratuitous, and limits the number of messages we feel comfortable sending.

Could this be changed to batch multiple events into a single request?  E.g. a background sender on a thread, firing off a fixed number of requests at a time, with potentially many log messages being enqueued and sent in one go.

FWIW this is how Logentries and other log client jars do it.

**Comment by marshallpierce** [comment]
I also was unhappy with the connection-per-request model, and rewrote much of the internals of the client here: https://github.com/truevault/rollbar-java. It uses non-blocking i/o and caches connections, so it may be more suitable for your needs. It doesn't handle batching of messages for you (though maybe it should), but there is a good way to do that yourself: implement a custom https://github.com/truevault/rollbar-java/blob/master/rollbar-http/src/main/java/com/truevault/rollbar/http/HttpItemClient.java that wraps the provided one that does batching how you want.

**Comment by proshin-roman** [comment]
I faced the same issue: each call to the `log` method produces a new HTTP connections. Finally a lot of server's threads have been blocked by the Rollbar appender. I think this is "must have" feature for the appender - use a pool of connections and send batch payloads. As well it would be nice to have async sending of entries.

**Comment by ThomasSolti** [comment]
Yeah, I fully agree with Roman! Please add this feature! Otherwise we also have this:
`http-nio-8080-exec-256" #217515 daemon prio=5 os_prio=0 tid=0x00007fe8ac10a800 nid=0x403 waiting for monitor entry [0x00007fe878bec000]
   java.lang.Thread.State: BLOCKED (on object monitor)
        at org.apache.log4j.Category.callAppenders(Category.java:204)
        - waiting to lock <0x0000000086a30e90> (a org.apache.log4j.spi.RootLogger)
        at org.apache.log4j.Category.forcedLog(Category.java:391)
        at org.apache.log4j.Category.log(Category.java:856)
        at org.slf4j.impl.Log4jLoggerAdapter.info(Log4jLoggerAdapter.java:305)`

**Comment by basoko** [comment]
Since version 1.0.0 there is a `BufferedSender` that uses a queue and a scheduled thread to send the different payloads. It is used by default in the android integration, so hope this can help.

**Comment by rokob** [comment]
This is possible by configuring the sender to use the BufferedSender. We cannot send a batch of items at once due to the API not operating that way.

**Comment by latos** [comment]
The issue should remain open until the API is updated to support batching - like most other APIs do

**Comment by rokob** [comment]
The API had private batch support but was removed for a variety of reasons. It won’t be coming back.
